ios/RNCAsyncStorage.m

```diff
@@ -269,7 +284,13 @@ - (instancetype)init
   if (!(self = [super init])) {
     return nil;
   }
-  RCTStorageDirectoryMigrationCheck();
+
+  // First migrate our deprecated path "Documents/.../RNCAsyncLocalStorage_V1" to "Documents/.../RCTAsyncLocalStorage_V1"
+  RCTStorageDirectoryMigrationCheck(RCTCreateStorageDirectoryPath_deprecated(RCTOldStorageDirectory), RCTCreateStorageDirectoryPath_deprecated(RCTStorageDirectory), YES, YES);
+  
+  // Then migrate what's in "Documents/.../RCTAsyncLocalStorage_V1" to "Application Support/[bundleID]/RCTAsyncLocalStorage_V1"
+  RCTStorageDirectoryMigrationCheck(RCTCreateStorageDirectoryPath_deprecated(RCTStorageDirectory), RCTCreateStorageDirectoryPath(RCTStorageDirectory), NO, NO);
```

Comment:
Since you're passing `YES, YES` to first migration,  then `NO, NO` to the second one, you can actually use one parameter for that, right?

Reply:
We could, but they represent pretty different actions. Also it sounds like we're going to drop RNC* migration support soon anyway so this leads us to a pretty easy plan there (drop the last parameter and keep the other one)